aa-schnorr-multisig-sdk/src/signers/Schnorrkel.ts

```diff
@@ -16,17 +16,42 @@ import {
   _verifyHash,
   _multiSigSignHash,
   _restorePublicNonces,
+  _hashMessage,
 } from "../core"
 import type { HashFunction, InternalNonces, InternalPublicNonces, InternalSignature } from "../core/types"
 import type { SignatureOutput } from "../types/signature"
 import { Challenge, FinalPublicNonce } from "../types/signature"
 
+// hash of private key + nonce
+const usedNonces: Set<string> = new Set()
+
 export class Schnorrkel {
   #nonces: Nonces = {}
 
+  private markNonceAsUsed(privateKey: Key, publicNonce: string): void {
+    const x = privateKey.buffer
+    const hash = _hashPrivateKey(x)
+    const hashPKplusNonce = hash + publicNonce
+    usedNonces.add(hashPKplusNonce)
+  }
+
+  private isNonceUsed(hashPK: string, publicNonce: string): boolean {
+    return usedNonces.has(hashPK + publicNonce)
+  }
+
+  resetUsedNonces(): void {
+    usedNonces.clear()
+  }
+
   private _setNonce(privateKey: Buffer): string {
     const { publicNonceData, privateNonceData, hash } = _generatePublicNonces(privateKey)
 
+    if (
+      this.isNonceUsed(hash, publicNonceData.kPublic.toString("hex")) ||
```

Comment:
This condition in practice is not possible to be true.
In `_generatePublicNonces` the nonces are generate using:
```
Buffer.from(ethers.randomBytes(32))
```
Expecting a collision at this point is as likely as expecting that the same private key is going to be generated twice.
Sure, this is not impossible in sense of laws of physics, but would take eons of trying to get to such situation